File: JAVA_COURSE/JAVA/10_1_Robot3/src/edu/javacourse/robot/RobotManager.java
package edu.javacourse.robot;

public class RobotManager
{
    public static void main(String[] args) {
        Robot robot = new RobotTotal(0, 0);
        System.out.println(robot.getClass().getName());
        robot.forward(20);
        robot.setCourse(90);
        robot.forward(20);
        robot.setCourse(90);
        robot.forward(50);
        // Напечатать координаты
        robot.printCoordinates();
        // Напечатать общую дистанцию
        //System.out.println(robot.getTotalDistance());
    }
}

File: JAVA_COURSE/KOTLIN/07_Robot1/src/KotlinRobot/RobotManager.kt
package KotlinRobot

object RobotManager {

    @JvmStatic
    fun main(args: Array<String>) {
        // Создаем объекта класса JavaRobot.Robot - теперь с параметрами
        val robot = Robot(20.0, 20.0)

        // Вперед на 20 метров
        robot.forward(20)
        // Напечатать координаты
        robot.printCoordinates()

        // Это более корректный способ менять курс.
        // Реализация внутри робота не сильно отличается, но
        // мы в любой момент сможем сделать более продвинутую версию
        // Но класс JavaRobot.RobotManager об этом даже не узнает
        robot.course = 90.0
        // Вперед на 20 метров
        robot.forward(20)
        // Напечатать координаты
        robot.printCoordinates()

        // Курс 45 градусов
        robot.course = 45.0
        // Вперед на 20 метров
        robot.forward(20)
        // Напечатать координаты
        robot.printCoordinates()
    }
}
